File: Modules/HumanResource/EmployeeAttendance/EmployeeAttendance/EmployeeAttendanceForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.EmployeeAttendance.Forms;

[FormScript("EmployeeAttendance.EmployeeAttendance")]
[BasedOnRow(typeof(EmployeeAttendanceRow), CheckNames = true)]
public class EmployeeAttendanceForm
{
    [OneThirdWidth]
    public DateTime AuthenticationDate { get; set; }

    [OneThirdWidth]
    public TimeSpan AuthenticationTime { get; set; }

    [OneThirdWidth]
    public int AuthenticationSecond { get; set; }
    [HalfWidth]
    public int EmployeeRowID { get; set; }
    [HalfWidth]
    public string PersonName { get; set; }

    public string DeviceName { get; set; }
    public string DeviceSerial { get; set; }
    public string CardNo { get; set; }
   

}

File: Modules/ServerTypes/EmployeeAttendance/EmployeeAttendanceForm.ts
import { DateEditor, StringEditor, IntegerEditor, LookupEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface EmployeeAttendanceForm {
    AuthenticationDate: DateEditor;
    AuthenticationTime: StringEditor;
    AuthenticationSecond: IntegerEditor;
    EmployeeRowID: LookupEditor;
    PersonName: StringEditor;
    DeviceName: StringEditor;
    DeviceSerial: StringEditor;
    CardNo: StringEditor;
}

export class EmployeeAttendanceForm extends PrefixedContext {
    static formKey = 'EmployeeAttendance.EmployeeAttendance';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!EmployeeAttendanceForm.init)  {
            EmployeeAttendanceForm.init = true;

            var w0 = DateEditor;
            var w1 = StringEditor;
            var w2 = IntegerEditor;
            var w3 = LookupEditor;

            initFormType(EmployeeAttendanceForm, [
                'AuthenticationDate', w0,
                'AuthenticationTime', w1,
                'AuthenticationSecond', w2,
                'EmployeeRowID', w3,
                'PersonName', w1,
                'DeviceName', w1,
                'DeviceSerial', w1,
                'CardNo', w1
            ]);
        }
    }
}